.github/workflows/awesome_workflow.yml

```diff
@@ -94,12 +94,26 @@ jobs:
     name: Compile checks
     runs-on: ${{ matrix.os }}
     needs: [MainSequence]
+    permissions:
+      pull-requests: write
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest, macOS-latest]
     steps:
       - uses: actions/checkout@v3
         with:
           submodules: true
-      - run: cmake -B ./build -S .
-      - run: cmake --build build
+      - run: |
+          cmake -B ./build -S .
+          cmake --build build
+      - name: Label on PR fail
+        uses: actions/github-script@v6
+        if: ${{ failure() && matrix.os == 'ubuntu-latest' && github.event_name == 'pull_request' }}
```

Comment:
I wonder if it makes sense to have a separate scheduled CI/Action which checks for failures on _**all**_ of the CI's instead of just checking for `failure` on this Action?

Reply:
The only one that checks and builds the code is this workflow. The other ones do not need to be checked, IMO.

Reply:
We would still find it easier to work with if we notice a label whenever *any* of the CI fails. I think its worth investigating 😄

Reply:
Yeah, that'd be better. Not 100% sure how would that be done, though.

Reply:
Ok, I think that is not something that could be done easily, lets merge this first and then continue investigating how to achieve that.

Reply:
Should I make an issue for that, or do you want to do it? 🤔

Reply:
Looking into it, I don't think there is an easy way to do this - I might open a feature request for this particular thing but we can't expect any result anytime soon 😅 Lets leave opening an issue for now - I'll open one when I know an api exists with this functionality.